include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
+    /// temporary made public
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getModulesFn()
+    std::string get_modules_fn();
 #endif
-    const std::string & getRevision() const;
-    int getAge() const;
-
-    /**
-    * @brief Ask for additional repository metadata type to download
-    *
-    * given metadata are appended to the default metadata set when repository is downloaded
-    *
-    * @param metadataType metadata type (filelists, other, productid...)
-    */
-    void addMetadataTypeToDownload(const std::string &metadataType);
-
-    /**
-    * @brief Stop asking for this additional repository metadata type
-    *
-    * given metadata_type is no longer downloaded by default
-    * when this repository is downloaded.
-    *
-    * @param metadataType metadata type (filelists, other, productid...)
-    */
-    void removeMetadataTypeFromDownload(const std::string &metadataType);
-
-    /**
-    * @brief Return path to the particular downloaded repository metadata in cache
-    *
-    * @param metadataType metadata type (filelists, other, productid...)
-    *
-    * @return file path or empty string in case the requested metadata does not exist
-    */
-    std::string getMetadataPath(const std::string &metadataType);
-
-    /**
-    * @brief Return content of the particular downloaded repository metadata
-    *
-    * Content of compressed metadata file is returned uncompressed
-    *
-    * @param metadataType metadata type (filelists, other, productid...)
-    *
-    * @return content of metadata file or empty string in case the requested metadata does not exist
-    */
-    std::string getMetadataContent(const std::string &metadataType);
-
-    /**
-    * @brief Mark whatever is in the current cache expired.
-    *
-    * This repo instance will alway try to fetch a fresh metadata after this
-    * method is called.
-    */
+
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getRevision()
+    const std::string & get_revision() const;
+
+    /// Gets age of the local cache
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getAge()
+    int64_t get_age() const;
+
+    /// Ask for additional repository metadata type to download.
+    /// Given metadata are appended to the default metadata set when repository is downloaded
+    /// @param metadataType metadata type (filelists, other, productid...)
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.addMetadataTypeToDownload(const std::string &metadataType)
+    void add_metadata_type_to_download(const std::string & metadata_type);
+
+    /// Stop asking for this additional repository metadata type.
+    /// Given metadata_type is no longer downloaded by default when this repository is downloaded.
+    /// @param metadataType metadata type (filelists, other, productid...)
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.removeMetadataTypeFromDownload(const std::string &metadataType)
+    void remove_metadata_type_from_download(const std::string & metadata_type);
```

Comment:
There is no getter to see what is in the list. What about to add it?

Reply:
Please could you add todo? I believe that the logic here is not correct, because it modifies only additional metadata.

Reply:
Ok. I will add TODO.

Reply:
In docs is. "Stop asking for this **additional** repository metadata type."
So, it works only with additional metadata. Basic set cannot be removed by this now.
 I added TODO.